src/JuMP.jl

```diff
@@ -73,7 +73,15 @@ An abstract type that should be subtyped for users creating JuMP extensions.
 """
 abstract type AbstractModel end
 
-mutable struct Model <: AbstractModel
+"""
+    value_type(::Type{<:Union{AbstractModel,AbstractVariableRef}})
+
+Return the return type of [`value`](@ref) for variables of that model. It
+defaults to `Float64` if it is not implemented.
+"""
+value_type(::Type{<:AbstractModel}) = Float64
+
+mutable struct GenericModel{T} <: AbstractModel
```

Comment:
Can we use a more descriptive name instead of `T`

Reply:
Maybe it's enough if it's in the docstring ? Having a small name might make the user thing there is a simple interpretation and no need to look up the docstring. At least with `T` it forces the user to read the docstring and get the full explanation and not get a wrong understanding ^^